webapp/channels/src/components/channel_settings_modal/channel_settings_access_rules_tab.tsx

```diff
@@ -160,6 +166,34 @@ function ChannelSettingsAccessRulesTab({
         setAutoSyncMembers((prev) => !prev);
     }, [parentPolicyAutoSync, expression, autoSyncMembers]);
 
+    // Validate that current user satisfies the expression
+    const validateSelfExclusion = useCallback(async (testExpression: string): Promise<boolean> => {
+        if (!testExpression.trim() || !currentUser?.id) {
+            return true; // No expression or no user, skip validation
+        }
+
+        try {
+            // Test if current user matches the expression
+            const result = await actions.searchUsers(testExpression, '', '', 100);
+            if (result.data) {
```

Comment:
If this is false and there are no errors, the function returns `true` from line 194, but I believe it should return false because if no user matched the expression, neither did the current user.

Reply:
you are right, now properly returns `false` when `result.data.users.length === 0`